bwapy/libbwapy.pyx

```diff
@@ -0,0 +1,356 @@
+# cython: language_level=3
+
+from pathlib import Path
+from typing import List
+
+from cpython cimport PyBytes_Check, PyUnicode_Check
+from libc.stdint cimport uint8_t
+from libc.stdio cimport SEEK_SET
+from libc.stdlib cimport calloc, free
+from libc.string cimport strncpy, strcat
+from pysam import FastxRecord, AlignmentFile, AlignedSegment
+
+__all__ = [
+    "BwaAlnOptions",
+    "BwaAlnOptionsBuilder",
+    "BwaIndex",
+    "BwaAln",
+]
+
+cdef class BwaAlnOptions:
+    """The container for options for [`BwaAln`][bwapy.BwaAln].
+
+    Use [`BwaAlnOptionsBuilder`][bwapy.BwaAlnOptionsBuilder] to use and set custom options.
```

Comment:
Why is a separate builder class necessary?

Reply:
Is the alternative is to have a `@property` with a setter for each option directly on `BwaAlnOptions`?  I suppose I am breaking the builder pattern by allowing the caller to pass in a `BwaAlnOptions` to the builder itself and re-use that object.  I am open to suggestions.